### PR TITLE
state parameter updates

### PR DESCRIPTION
**Pull request by tamigoodall**

### CODE REVIEW

**Comment by tamigoodall** [comment]
My first one!!!

File: articles/protocols/oauth2/oauth-state.md [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
 
-By using the state parameter to hold a value for verification, malicious requests can be denied.
+## How to use the `state` parameter
 
-![](/media/articles/protocols/CSRF_Diagram.png)
+Deny malicious requests by setting the `state` parameter to hold a value for verification.
 
-::: note
-Depending on the application type or framework this may be included for the developer. Also the exact structure of the requests may differ.
+::: Note
```
Comment (mpaktiti) [inline articles/protocols/oauth2/oauth-state.md]: I deployed locally to test the doc (the heroku review app refuses to work) and this displays like that:

image

Looks like the `::: note` syntax is case sensitive. Change and see if it gets fixed? (spoiler alert: it does :P)

File: articles/protocols/oauth2/oauth-state.md [hunk 2]
```diff
@@ -1,24 +1,26 @@
...
-For the most basic cases the `state` parameter should be a [nonce](https://en.wikipedia.org/wiki/Cryptographic_nonce) as shown in the example below.  **But this field can also be a [Base64](https://en.wikipedia.org/wiki/Base64) encoded json object that can hold multiple values [such as a return URL](/tutorials/redirecting-users).**
+![](/media/articles/protocols/CSRF_Diagram.png)
 
-## How to use the state parameter
+::: note
+Depending on the application type or framework, this may be included for the developer. Also the exact structure of the requests may differ.
+:::
 
-By using the state parameter to hold a value for verification, malicious requests can be denied.
+## How to use the `state` parameter
```
Comment (mpaktiti) [inline articles/protocols/oauth2/oauth-state.md]: Can you remove please the code formatting from the header? It should be `## How to use the state parameter`

File: articles/protocols/oauth2/oauth-state.md [hunk 3]
```diff
@@ -1,24 +1,26 @@
...
-![](/media/articles/protocols/CSRF_Diagram.png)
+Deny malicious requests by setting the `state` parameter to hold a value for verification.
 
-::: note
-Depending on the application type or framework this may be included for the developer. Also the exact structure of the requests may differ.
+::: Note
+For most cases, the `state` parameter should be a [nonce](https://en.wikipedia.org/wiki/Cryptographic_nonce) as shown in the example below.  **But this field can also be a [Base64](https://en.wikipedia.org/wiki/Base64) encoded json object that can hold multiple values [such as a return URL](/tutorials/redirecting-users).**
 :::
 
 1. Before redirecting a request to the [IdP](/identityproviders), have the application generate a random string.
```
Comment (mpaktiti) [inline articles/protocols/oauth2/oauth-state.md]: In step 2, the URL is broken. The correct is `/security/store-tokens#web-storage-local-storage-session-storage-`

File: articles/protocols/oauth2/oauth-state.md [hunk 4]
```diff
@@ -1,24 +1,26 @@
 ---
-title: The State Parameter
+title: State Parameter
 description: Explains how to use the state parameter in authentication requests to help prevent CSRF attacks.
 ---
 
-# The State Parameter
+# State Parameter
 
 The `state` parameter is an authentication parameter used to help mitigate [CSRF attacks](https://en.wikipedia.org/wiki/Cross-site_request_forgery).
```
Comment (mpaktiti) [inline articles/protocols/oauth2/oauth-state.md]: I think this needs more info. The "is an authentication parameter" is a bit generic, we don't even say what's the protocol that's using it. Or how this parameter actually helps against CSFR.

Maybe you could also use this as input https://community.auth0.com/t/usage-of-the-oauth-state-parameter/8266 (but write it in a more understandable format)

**Review by mpaktiti — CHANGES_REQUESTED** [review]
Tami, good work :) I left some comments, let me know if they are not clear or you want to discuss them.

**Comment by mpaktiti** [comment]
@tamigoodall Once you are done with pushing the changes remove the `update-needed` label and this will pop right back to my to-review list. Thanks!

File: articles/protocols/oauth2/oauth-state.md (hunk 1 above)
Comment (tamigoodall) [reply articles/protocols/oauth2/oauth-state.md]: Makes sense.

**Review by tamigoodall — COMMENTED** [review]
Incorporated all changes

**Comment by tamigoodall** [comment]
@mpaktiti When I click **See review**, it doesn't do anything. I can't see the changes.

**Comment by tamigoodall** [comment]
When I click **See review** next to your review, it jumps to the top of this page. Not sure what your change request is.

**Comment by mpaktiti** [comment]
@tamigoodall Since the time you started working on this (apologies on the delay to merge) we were notified of a second use case. So I pushed some changes in this PR to incorporate these changes and restructure the headers a bit. I'll be merging within the day. Thanks!

**Pull request merged**